src/query/config/src/outer_v0.rs

```diff
@@ -1258,11 +1258,25 @@ impl From<InnerHiveCatalogConfig> for HiveCatalogConfig {
     }
 }
 
+#[derive(serde::Serialize, serde::Deserialize, Debug, PartialEq, Eq)]
+pub enum MetaType {
+    #[serde(rename = "remote")]
+    Remote,
+
+    #[serde(rename = "embedded")]
+    Embedded,
+}
+
 /// Meta config group.
 /// TODO(xuanwo): All meta_xxx should be rename to xxx.
 #[derive(Clone, PartialEq, Eq, Serialize, Deserialize, Args)]
 #[serde(default)]
 pub struct MetaConfig {
+    /// MetaStore type: remote or embedded
+    #[clap(long = "meta-type", default_value = "embedded")]
+    #[serde(alias = "meta_type")]
+    pub meta_type: String,
```

Comment:
Is it possible to use `pub meta_type: MetaType` to avoid manual conversion?

Reply:
> Is it possible to use `pub meta_type: MetaType` to avoid manual conversion?

This manual conversion is by design. We will maintain the forward compatibility by implementing `try_into` from the outer config to the inner config. This way, we can do anything to `MetaType` without worrying about breaking our users' config.

Reply:
make sense